composites/AlgoliaSearch/AlgoliaSearcher.js

```diff
@@ -0,0 +1,384 @@
+		this.setState( {
+			results: results,
+			errorMessage: "",
+			searching: false,
+		} );
+	}
+
+	/**
+	 * Performs a search with the searchstring saved in the state and sets the
+	 * results property of the state to the results found.
+	 *
+	 * @returns {void}
+	 */
+	updateSearchResults() {
+		this.setState( { searching: true } );
+
+		this.performSearch( this.state.searchString ).then( this.processResults.bind( this ) ).catch( this.processSearchError.bind( this ) );
+	}
+
+	/**
+	 * Performs a search with a given search string on the algolia index which
+	 * information was passed in the AlgoliaSearcher's props.
+	 *
+	 * @param {string} searchString The words or sentence to get the results for.
+	 *
+	 * @returns {Promise} The promise that is performing the search.
+	 */
+	performSearch( searchString ) {
+		return new Promise( ( resolve, reject ) => {
+			this.index.search( searchString, ( err, data ) => {
+				if ( err ) {
+					reject( err );
+					return;
+				}
+
+				resolve( data.hits );
+			} );
+		} );
+	}
+
+	/**
+	 * Sets all values required to display the detail view of a search result.
+	 *
+	 * @param {number} resultIndex The index of the article you want to
+	 *                                  show in the state.results array.
+	 * @returns {void}
+	 */
+	showDetail( resultIndex ) {
+		let usedQueries = this.state.usedQueries;
+		let post = this.state.results[ resultIndex ];
+
+		usedQueries[ this.state.searchString ][ post.objectID ] = {
+			title: post.post_title,
+			link: post.permalink,
+		};
+
+		this.setState( {
+			showDetail: resultIndex,
+			usedQueries: usedQueries,
+		} );
+	}
+
+	/**
+	 * Hides the details page and return to the results page.
+	 *
+	 * @returns {void}
+	 */
+	hideDetail() {
+		this.setState( { showDetail: false } );
+	}
+
+	/**
+	 * Renders a no results found text.
+	 *
+	 * @returns {ReactElement} The no results found text.
+	 */
+	renderNoResultsFound() {
+		let searchResultContent = <p>{ this.props.noResultsText }</p>;
+		a11ySpeak( this.props.noResultsText );
+
+		return searchResultContent;
+	}
+
+	/**
+	 * Maps the results to SearchResult components.
+	 *
+	 * @param {object} results The results returned by Algolia.
+	 *
+	 * @returns {void}
+	 */
+	resultsToSearchItem( results ) {
+		return results.map( ( result, index ) => {
+			return <SearchResult
+				key={ result.objectID }
+				post={ result }
+				showDetail={ this.showDetail.bind( this, index ) }
+			/>;
+		} );
+	}
+
+	/**
+	 * Renders the search results list.
+	 *
+	 * @returns {ReactElement} A div with either the search results, or a div with a message that no results were found.
+	 */
+	renderSearchResults() {
+		let searchResultContent;
+		let resultsCount = this.state.results.length;
+
+		// We'll check to see whether no results are returned.
+		if ( resultsCount <= 0 && this.state.searchString !== "" ) {
+			return this.renderNoResultsFound();
+		}
+
+		if ( resultsCount > 0 ) {
```

Comment:
I would check for results being 0 and return the `searchResultContent ` and do the rest of it outside the if.
```
if ( resultsCount === 0 ) {
    return searchResultContent;
}

// The other things.